### PR TITLE
Feat api functionalities

### PR DESCRIPTION
**Pull request by devananth**
## Completed P0 features:

    - Added weather api 
    - Added random quotes
    - Added random wallpaper

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -7,8 +8,26 @@ function App() {
     dataState: { userName },
   } = useData();
 
+  const [imageUrl, setImageUrl] = useState(null);
+
+  useEffect(() => {
+    fetch(
+      `https://api.unsplash.com/photos/random/?client_id=XqByhaz8qx069KZ_7Kk528zhO_02RDftaH3ioB8y1No&&orientation=landscape&&query=mountains%20dark`
```
Comment (ArjunGTX) [inline src/App.js:15]: Can store the base URL in a variable and the API key in `.env` file

File: src/components/Weather.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from "react";
+
+const Weather = () => {
+  const [location, setLocation] = useState({ latitude: null, longitude: null });
+  const [weatherData, setWeatherData] = useState(null);
+
+  const getWeatherData = async () => {
+    const { latitude, longitude } = location;
+    // try {
```
Comment (ArjunGTX) [inline src/components/Weather.jsx:9]: remove commented code

File: src/components/Weather.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    //     const data = await response.json();
+
+    //     return data;
+    //   }
+    // } catch (error) {
+    //   console.log("Something went wrong");
+    // }
+
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=2728244c87e645e190765ebb90367589`
```
Comment (ArjunGTX) [inline src/components/Weather.jsx:24]: Same as mentioned in first comment

File: src/App.js [hunk 4]
```diff
@@ -7,8 +8,26 @@ function App() {
...
+    )
+      .then((res) => res.json())
+      .then((x) => {
+        console.log(x);
```
Comment (shivanigangadharan) [inline src/App.js:19]: can remove console logs wherever not reqd

File: src/components/Weather.jsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    //       `https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=2728244c87e645e190765ebb90367589`
+    //     );
+
+    //     const data = await response.json();
+
+    //     return data;
+    //   }
+    // } catch (error) {
+    //   console.log("Something went wrong");
+    // }
```
Comment (shivanigangadharan) [inline src/components/Weather.jsx:21]: pls remove unwanted comments

File: src/components/Weather.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          temperature: (data.main.temp - 273.15).toFixed(2),
+          icon: data.weather[0].icon,
+        };
+        console.log(value);
+        setWeatherData(value);
+      });
+  };
+
+  useEffect(() => {
+    navigator.geolocation.getCurrentPosition((position) => {
```
Comment (ArjunGTX) [inline src/components/Weather.jsx:40]: Just learned a new thing. Thanks

**Review by ArjunGTX — APPROVED** [review]

**Review by shivanigangadharan — APPROVED** [review]
Code is well structured, can merge this.

File: src/App.css [hunk 7]
```diff
@@ -7,6 +7,7 @@
   flex-direction: column;
   justify-content: center;
   align-content: center;
+  color : white;
```
Comment (aqsaaqeel) [inline src/App.css:10]: can use variable

**Review by aqsaaqeel — APPROVED** [review]
You can add a screenshot of the extension, the netlify preview is notworking

**Pull request merged**